### PR TITLE
Document TDR and SAM service url env vars (#2134)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**

### CODE REVIEW

File: environment.py [hunk 1]
```diff
@@ -370,5 +370,13 @@ def env() -> Mapping[str, Optional[str]]:
         #
         # tdr:broad-jade-dev-data:dataset/hca_mvp
         # tdr:broad-jade-dev-data:snapshot/hca_mvp
-        'AZUL_TDR_TARGET': None
+        'AZUL_TDR_TARGET': None,
+
+        # Instance of the Terra Data Repository API to use for retrieving
+        # snapshot or dataset descrptions.
```
Comment (hannes-ucsc) [inline environment.py:376]:
```suggestion
        # The URL of the Terra Data Repository instance to index metadata from.
```

and wrap as necessary

File: environment.py [hunk 2]
```diff
@@ -370,5 +370,13 @@ def env() -> Mapping[str, Optional[str]]:
...
+        'AZUL_TDR_SERVICE_URL': None,
+
+        # Instance of Broad's SAM tool to use for registering Google service
+        # accounts so that they may be authorized to access TDR.
```
Comment (hannes-ucsc) [inline environment.py:380]:
```suggestion
        # The URL of an instance of Broad Institute's SAM. This needs to point to the SAM instance that's used by the TDR instance configured in `AZUL_TDR_SERVICE_URL`.
```
wrap as necessary.

File: UPGRADING.rst [hunk 3]
```diff
@@ -10,6 +10,12 @@ branch that does not have the listed changes, the steps would need to be
 reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
+#2025 Register indexer SA with Broad's SAM during deployment
```
Comment (hannes-ucsc) [inline UPGRADING.rst:13]: Double check if this is really where you introduced these variables.

Comment (noah-aviel-dove) [reply UPGRADING.rst:13]: https://github.com/DataBiosphere/azul/commit/08e765f76f38c9764d4dee912b146d7861b30c5d#diff-ca6bc615c166eaf90c362521c5bdab0cR37-R38

**Comment by hannes-ucsc** [comment]
Skipping sandbox.

**Pull request merged**